app/controllers/sessions_controller.rb

```diff
@@ -29,4 +20,14 @@ def destroy
     flash[:notice] = "You are now logged out"
     redirect_to root_path
   end
+
+  private
+
+  def redirect_path
+    if previous_path == cart_path || previous_path == items_path
```

Comment:
What is a user clicked login from items path? I think they should go back to items. Right?